cmd/state-exec/main.go

```diff
@@ -65,10 +65,22 @@ func run() error {
 	}
 	logr.Debug("message data - pid: %s, exec: %s", hb.ProcessID, hb.ExecPath)
 
+	a, err := newAttempt()
+	if err != nil {
+		return fmt.Errorf("cannot create new attempt: %w", err)
+	}
+	logr.Debug("message data - exec: %s", a.ExecPath)
+
 	meta, err := newExecutorMeta(hb.ExecPath)
```

Comment:
The reason that the heartbeat type is being created early is to obtain the execpath. If, instead, the attempt type is constructed first, the execpath can be passed into the heartbeat construction func later (closer to where it is used - the debugging output should be able to be moved as well), and passed into the newExecutorMeta func here.